src/connection.rs

```diff
@@ -310,6 +339,13 @@ impl StreamMuxer for Connection {
                     // might have been closed/dropped
                     inbound_tx.send(data).ok();
                 }
+                SubstreamMessageType::OpenTimeout => {
+                    self.handle_timeout(&msg.substream_id);
+
+                    // TODO return error or just ignore since this is only a timeout for
+                    // opening substreams.
+                    return Poll::Ready(Err(Error::StreamOpenTimeout(msg.substream_id)));
```

Comment:
this shouldn't be `Poll::Ready` as there wasn't actually a stream muxer event, also returning an error here will cause the future to error, which will result in the connection being closed which is not what we want. I'd remove the `SubstreamMessageType::OpenTimeout` entirely